DataPreprocessing.ipynb

```diff
@@ -66,33 +70,45 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 31,
+   "execution_count": 13,
    "metadata": {
-    "collapsed": true
+    "collapsed": false
    },
    "outputs": [],
    "source": [
-    "def create_train_data(img_size):\n",
+    "def create_train_data(img_size, train_dir):\n",
     "    training_data = []\n",
-    "    for file in os.listdir(TRAIN_DIR):\n",
+    "    for file in os.listdir(train_dir):\n",
     "        print('Starting on file {}'.format(file))\n",
-    "        file_path = os.path.join(TRAIN_DIR,file)\n",
+    "        file_path = os.path.join(train_dir,file)\n",
+    "        \n",
     "        data_label = label_img(file)\n",
+    "        if data_label == [1,0,0]:\n",
+    "            label_folder = 'Type_1\\\\'\n",
+    "        elif data_label == [0,1,0]:\n",
+    "            label_folder = 'Type_2\\\\'\n",
+    "        else:\n",
+    "            label_folder = 'Type_3\\\\'\n",
+    "\n",
     "        file_len = len(os.listdir(file_path))\n",
     "        for i, img in enumerate(os.listdir(file_path)):\n",
     "            show_progress(i,file_len) # Print progress\n",
     "            label = data_label\n",
     "            path = os.path.join(file_path,img)\n",
-    "            img = cv2.resize(cv2.imread(path,cv2.IMREAD_GRAYSCALE), (img_size,img_size))\n",
-    "            training_data.append([np.array(img),np.array(label)])\n",
-    "    # So that we don't have to do this again, unless we change something here.\n",
-    "    np.save('train_data.npy', training_data)\n",
-    "    return training_data"
+    "            try:\n",
+    "                img = cv2.resize(cv2.imread(path,cv2.IMREAD_GRAYSCALE), (img_size,img_size))\n",
+    "            except:\n",
+    "                print('\\nError resizing image {} in {}'.format(img, file_path))\n",
+    "                continue\n",
+    "            training_data =(np.array(img),np.array(label))\n",
+    "            # So that we don't have to do this again, unless we change something here.\n",
```

Comment:
We don't need to shuffle if we're saving each image separately like we are here. Instead when we read images in, we do this:
```random.choices(os.listdir, k=batch_size)```